mmcv/ops/csrc/pytorch/bbox_overlaps_cuda.cu

```diff
@@ -8,8 +8,14 @@ void BBoxOverlapsCUDAKernelLauncher(const Tensor bboxes1, const Tensor bboxes2,
   int num_bbox1 = bboxes1.size(0);
   int num_bbox2 = bboxes2.size(0);
 
+#ifdef __NVCC__
   at::cuda::CUDAGuard device_guard(bboxes1.device());
   cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+#endif
+#ifdef __HIP_PLATFORM_HCC__
+  // at::cuda::HIPGuard device_guard(bboxes1.device());
```

Comment:
Why comment out HIPGuard?

Reply:
Because pytorch handles ROCm as CUDA, it will cause assert device type error if not comment.
I'm not familiar with CUDA or ROCm, but i hope ROCM can get official efficient support.

Reply:
It seems that here should use HIPGuardMasqueradingAsCUDA and getCurrentHIPStreamMasqueradingAsCUDA?

Reply:
It seems that here should use HIPGuardMasqueradingAsCUDA and getCurrentHIPStreamMasqueradingAsCUDA?